mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/impl/SetupStage.scala

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.mqtt.streaming.impl
+
+import akka.annotation.InternalApi
+import akka.stream._
+import akka.stream.scaladsl.{Flow, Keep, Sink, Source}
+import akka.stream.stage._
+
+import scala.concurrent.{Future, Promise}
+
+private final class SetupSinkStage[T, M](factory: ActorMaterializer => Attributes => Sink[T, M])
```

Comment:
It’d be great to have doc describing the motivation for these types.

Reply:
Soon to become part of Akka... https://github.com/akka/akka/pull/26477